proto/minder/v1/minder.proto

```diff
@@ -2626,3 +2626,42 @@ message EvaluationHistoryAlert {
     // the structure and contents are alert specific, and are subject to change.
     string details = 2;
 }
+
+message SelectorEntity {
+    // one of repository, pull_request, artifact (see oneof entity)
+    Entity entity_type = 1;
+    // the name of the entity, same as the name in the entity message
+    string name = 2;
+
+    oneof entity {
+        SelectorRepository repository = 5;
+        SelectorArtifact artifact = 6;
+    }
+
+    message SelectorRepository {
+        // the full name of the repository, e.g. stacklok/minder
+        string name = 1;
+        // the name of the repository, e.g. minder
+        string repo_name = 2;
+        // the owner of the repository, e.g. stacklok
+        string owner = 3;
+
+        // is_fork is true if the repository is a fork, nil if "don't know" or rather
+        // not applicable to this provider
+        optional bool is_fork = 4;
+        // is_private is true if the repository is private, nil if "don't know" or rather
+        // not applicable to this provider
+        optional bool is_private = 5;
+    }
+
+    message SelectorArtifact {
```

Comment:
Sorry, did you mean if we need the `is_private` attribute of the `SelectorRepository` message or if we should add `is_private` to the `SelectorArtifact` message?

Reply:
I meant that we might want to be able to distinguish between public and private artifacts, as well as public and private repos.  But this may tie in to the [key-values from Provider comment](https://github.com/stacklok/minder/pull/3797#discussion_r1669286088).

Reply:
added! I would like to tie the k-v map in a follow up, though.